cmd/core/main.go

```diff
@@ -235,7 +235,7 @@ func registerHealthChecks(mgr ctrl.Manager) error {
 	if err := mgr.AddReadyzCheck("ping", healthz.Ping); err != nil {
 		return err
 	}
-
+	// TODO: change the health check to be different from readiness check
```

Comment:
I think it could be a good first issue if you have more detailed design about how do we define the controller is readiness.

Reply:
good idea